Extract Enabled Obsolete Operating Systems From LDAP

Shad0wC0ntr0ller opened the pull request:
Extract Enabled Obsolete Operating Systems From LDAP

cme ldap 192.168.199.231 -u 'user' -p 'password' -M obsolete


![image](https://github.com/mpgn/CrackMapExec/assets/90877534/27164819-487a-4cd5-8b6d-e8e41a2a286f)

This module will need some slight readjusting based off of the tips you have provided me on previous modules. I have had this one in the stash for a minute without much editing so I will handle that shortly. Also can you please let me know where is the preferred folder to save things, for example this module will save the dnshostname to a file in /tmp/obsoletehosts.txt. where would you actually like this saved

NeffIsBack commented:
Check out the .cme folder in your home directory. The wcc module saves its results as logs in the logs directory inside .cme with the option to export the results as well. Maybe something like that would be an option.

For me a nice addition for this module would be to also retrieve the ip address for the systems if that's possible

Shad0wC0ntr0ller commented:
Great, I will take a look at changing the save location and adding the option for the ip address if it is available. TY

Shad0wC0ntr0ller commented:
Good Morning, I cleaned up the old code. I changed the save location, switched .info to .display, and added a part to get the ip address if available. Please let me know if there is anything else that needs to be changed.

![image](https://github.com/mpgn/CrackMapExec/assets/90877534/767724de-c738-4a01-b4c9-66b21b38765e)

Shad0wC0ntr0ller commented:
Just to add, in about 30 days we will need to add Windows Server 2012 and Windows Server 2012 R2

"Windows Server 2012 and Windows Server 2012 R2 will end on October 10, 2023. After this date, these products will no longer receive security updates, non-security updates, bug fixes, technical support, or online technical content updates."

https://learn.microsoft.com/en-us/lifecycle/announcements/windows-server-2012-r2-end-of-support

Other than that @mpgn and @NeffIsBack this module is ready for review. Thank you
